src/components/type-group/TypeGroup.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './TypeGroup.css';
+
+class TypeGroup extends Component {
+    render() {
+        const { className, name } = this.props;
+
+        const children = React.Children.map(this.props.children, (child) => cloneElement(child, {
+            groupName: name,
```

Comment:
I would add a `selected` **boolean** prop to TypeOption and a `selected` **string** prop to TypeGroup. This way, we can control the selected option directly in the group.

This implies removing the existing `defaultSelected` because `selected` gives us more control over the component.